lib/cosmos/interfaces/interface.rb

```diff
@@ -118,37 +118,70 @@ def initialize
     # Connects the interface to its target(s). Must be implemented by a
     # subclass.
     def connect
-      raise "Interface connect method not implemented"
     end
 
     # Indicates if the interface is connected to its target(s) or not. Must be
     # implemented by a subclass.
     def connected?
-      raise "Interface connected? method not implemented"
+      false
     end
 
     # Disconnects the interface from its target(s). Must be implemented by a
     # subclass.
     def disconnect
-      raise "Interface disconnect method not implemented"
     end
 
-    # Retrieves the next packet from the interface. Must be implemented by a
-    # subclass.
+    # Retrieves the next packet from the interface.
+    # @return [Packet] Packet constructed from the data. Packet will be
+    #   unidentified (nil target and packet names)
     def read
```

Comment:
I implemented read such that it calls read_data which is implemented by a module which is extended into the Interface. That way I can call the post_read_data and post_read_packet methods consistently along with incrementing bytes_read and read_count.